RIP-3 Nft Bank/RIP-3 Nft Bank.md

```diff
@@ -0,0 +1,85 @@
+
+Functions:
+
+Note: All mutative functions will have a modifier that requires that either msg.sender == the affected account's address or that the current blockNumber - lastAccessed variable is over a certain time frame and that msg.sender == one of the backup addresses. And all mutative functions will reset the lastAccessed blockNumber of an address to the current blockNumber, but not if the functions were accessed by the backup addresses.
+
+-   depositBatch function to deposit multiple assets. This function takes in an array of assetData structs, an array of uint256 deposit amounts and an address.
+    -   checks if the user possesses the amount of tokens they want to deposit.
+    -   calls the deposit function in BankStorage.sol to update the storedAmounts mapping.
+    -   calls a transfer function to send the item to Bank.sol.
+-   withdrawBatch function to withdraw multiple assets. This function takes in an array of assetData structs, an array of uint256 withdraw amounts and an address.
+    -   requires that stored amount of the item is greater than or equal to the amount to withdraw.
+    -   calls the withdraw function in BankStorage.sol to update the storedAmounts mapping.
+    -   calls a transfer function to retrieve the item from the bank.
+-   addBackup function to add a back up address in case one loses their keys.
+-   removeBackup function to remove an address from the backupAddresses array.
+-   setPassword function which sets an account password required to lock and unlock an account.
+-   lock function which requires a password and signature to set the accountLocked boolean to true or false.
+-   tokenAmount function to call the tokenAmount function in BankStorage.sol
+-   hashAssetId internal function which creates a hash of the item's contract address and token Id.
+
+#### BankStorage.sol
+Mappings:
+-   (bytes4 assetIdHash => uint256 tokenAmount) storedAmounts 
+-   (address user => storedAmounts) assetInfo
+
+Functions:
+-   withdraw function which decreases the value of tokenAmount in the storedAmounts mapping.
+-   deposit function which increases the value of tokenAmount in the storedAmounts mapping.
+-   tokenAmount function to query the amount of an item stored in the Bank by the caller.
+
+## Rationale
+Design Considerations:
+-   In BankStorage.sol, because storedAmounts, which basically is the bank account, is tied to an account address, backup addresses will not be able to obtain true admin priveleges. To remedy this, it is possible that instead of mapping addresses to storedAmounts, we can map uint256 accountNumbers to the storedAmounts mapping. And in Bank.sol, we can add a createAccount function, which initializes the SecurityInfo struct of an accountNumber with an adminAddress variable. So if a user's private keys are lost, instead of checking whether msg.sender is equal to one of the backup addresses, we can set one of the backup addresses as the new admin address of the account.
```

Comment:
I don't think this is necessary anymore because we made it so that the NFT bank is per-user now.